### PR TITLE
Add initial operatorhub templates

### PR DESCRIPTION
**Pull request by tylerauerbeck**
#### What is this PR About?
This adds some initial templates for interacting with OperatorHub. This assumes that OLM is running and configured appropriately.

#### How do we test this?

Test by deploying the Elasticsearch operator on OCP 4:

```
oc new-project my-namespace

oc process -f operatorgroup.yaml -p NAME=openshift-logging-operatorgroup -p TARGET_NAMESPACE=my-namespace| oc apply -f - -n my-namespace

oc process -f catalogsourceconfig.yml -p NAME=elasticsearch-operator -p OPERATORS="elasticsearch-operator" | oc apply -f - -n openshift-marketplace

oc process -f subscription.yaml -p NAME=elasticsearch-operator -p OPERATOR_NAME=elasticsearch-operator -p OPERATOR_NAMESPACE=openshift-operators -p CATALOG_SOURCE_NAME=elasticsearch-operator | oc apply -f - -n openshift-operators
```

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

File: operatorhub/subscription.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+message: |-
+  The following subscription has been created: ${NAME}
+metadata:
+  name: subscription-template
+objects:
+- apiVersion: operators.coreos.com/v1alpha1
+  kind: Subscription
+  metadata:
+    name: "${NAME}"
+    namespace: "${OPERATOR_NAMESPACE}"
```
Comment (pabrahamsson) [inline operatorhub/subscription.yml]: I think this line should be removed and instead set the namespace from cli or from the applier.

File: operatorhub/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# OperatorHub
+
+The below templates are used to deploy Operators from OperatorHub without having to interact with the WebUI.
+
+To do this you would create the following in order:
+
+- [OperatorGroup](operatorgroup.yml)
+- [CatalogSourceConfig](catalogsourceconfig.yml)
+- [Subscription](subscription.yml)
```
Comment (pabrahamsson) [inline operatorhub/README.md]: Since the order matter here, can you make it an ordered list?

**Review by pabrahamsson — CHANGES_REQUESTED** [review]
Looks good @tylerauerbeck, two minor comments inline.

**Comment by tylerauerbeck** [comment]
@pabrahamsson Fixed. Let me know what you think.

**Review by pabrahamsson — APPROVED** [review]
LGTM

**Pull request merged**